From: Julian Andres Klode <jak@debian.org>
Date: Tue, 15 Sep 2015 16:02:03 +0200
Subject: Disable support for drivers and live patch

Disable support for drivers, as this is somewhat Ubuntu specific.
---
 data/gtkbuilder/main.ui                         |  5 ++++-
 softwareproperties/gtk/SoftwarePropertiesGtk.py | 11 +++++++++--
 2 files changed, 13 insertions(+), 3 deletions(-)

diff --git a/data/gtkbuilder/main.ui b/data/gtkbuilder/main.ui
index 146a9ed..3000d76 100644
--- a/data/gtkbuilder/main.ui
+++ b/data/gtkbuilder/main.ui
@@ -775,7 +775,7 @@
                 </child>
                 <child>
                   <object class="GtkAlignment" id="alignment15">
-                    <property name="visible">True</property>
+                    <property name="visible">False</property>
                     <property name="can_focus">False</property>
                     <property name="left_padding">12</property>
                     <child>
@@ -1026,6 +1026,7 @@
                 <property name="tab_fill">False</property>
               </packing>
             </child>
+            <!--
             <child>
               <object class="GtkBox" id="vbox_drivers">
                 <property name="visible">True</property>
@@ -1123,6 +1124,7 @@
                 <property name="tab_fill">False</property>
               </packing>
             </child>
+                        -->
             <child>
               <object class="GtkBox" id="dev_options">
                 <property name="visible">True</property>
@@ -1187,6 +1189,7 @@
                 <property name="tab_fill">False</property>
               </packing>
             </child>
+            -->
           </object>
           <packing>
             <property name="expand">True</property>
diff --git a/softwareproperties/gtk/SoftwarePropertiesGtk.py b/softwareproperties/gtk/SoftwarePropertiesGtk.py
index 75ff85d..71f068d 100644
--- a/softwareproperties/gtk/SoftwarePropertiesGtk.py
+++ b/softwareproperties/gtk/SoftwarePropertiesGtk.py
@@ -61,7 +61,11 @@ import softwareproperties.distro
 from softwareproperties.SoftwareProperties import SoftwareProperties
 import softwareproperties.SoftwareProperties
 
-from UbuntuDrivers import detect
+try:
+    from UbuntuDrivers import detect
+except ImportError as e:
+    detect = None
+    logging.error("Cannot import UbuntuDrivers: " + str(e))
 
 if GLib.pyglib_version < (3, 9, 1):
     GLib.threads_init()
@@ -227,7 +231,7 @@ class SoftwarePropertiesGtk(SoftwareProperties, SimpleGtkbuilderApp):
 
     def on_main_notebook_page_switched(self, notebook, page, page_num):
         # On the additional drivers page, don't show the backend revert button.
-        if page == self.vbox_drivers:
+        if False and page == self.vbox_drivers:
             self.button_revert.set_visible(False)
             if not self.detect_called:
                 GLib.idle_add(lambda: threading.Thread(target=self.detect_drivers).start())
@@ -1175,6 +1179,8 @@ class SoftwarePropertiesGtk(SoftwareProperties, SimpleGtkbuilderApp):
     def init_drivers(self):
         """Additional Drivers tab"""
 
+        return
+
         self.button_driver_revert = Gtk.Button(label=_("Re_vert"), use_underline=True)
         self.button_driver_revert.connect("clicked", self.on_driver_changes_revert)
         self.button_driver_apply = Gtk.Button(label=_("_Apply Changes"), use_underline=True)
@@ -1520,6 +1526,7 @@ class SoftwarePropertiesGtk(SoftwareProperties, SimpleGtkbuilderApp):
             return False
 
     def is_livepatch_supported(self):
+        return False
         distro = aptsources.distro.get_distro()
         di = distro_info.UbuntuDistroInfo()
         return self.has_online_accounts() and \
